server/public/model/oauth_outgoing_connection.go

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package model
+
+import (
+	"net/http"
+	"unicode/utf8"
+)
+
+type OAuthOutgoingConnectionGrantType string
+
+func (gt OAuthOutgoingConnectionGrantType) IsValid() bool {
+	return gt == GrantTypeClientCredentials || gt == GrantTypePassword
+}
+
+const (
+	GrantTypeClientCredentials OAuthOutgoingConnectionGrantType = "client_credentials"
+	GrantTypePassword          OAuthOutgoingConnectionGrantType = "password"
+
+	defaultGetConnectionsLimit = 50
+)
+
+type OAuthOutgoingConnection struct {
```

Comment:
I used that thinking about the URL. I saw that the apps URL was `/oauth/apps` so it made sense to me that this would be `/oauth/outgoing_connections`, thus using `OAuthOutgoingConnection` was to follow that naming approach. 0/5 though.

Reply:
`OutgoingOAuthConnection` sounds better to me. The type name will be used on its own and is not strictly connected to the URL path. `OAuthOutgoingConnection` is just hard to read for humans.